dev/com.ibm.ws.security.jwtsso_fat/fat/src/com/ibm/ws/security/jwtsso/fat/ConfigAttributeTests.java

```diff
@@ -0,0 +1,483 @@
+    public static LibertyServer server;
+
+    private TestValidationUtils validationUtils = new TestValidationUtils();
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        ShrinkHelper.exportDropinAppToServer(server, CommonFatApplications.getTestMarkerApp());
+        ShrinkHelper.exportAppToServer(server, CommonFatApplications.getFormLoginApp());
+        serverTracker.addServer(server);
+        server.startServer();
+    }
+
+    /**
+     * Test the config attributes cookieName and includeLtpaCookie.
+     * Invoke app on the happy path, check that jwt cookie with correct name came back,
+     * and that ltpa cookie came back.
+     *
+     */
+    @Mode(TestMode.LITE)
+    @Test
+    public void test_cookieName_includeLtpa() throws Exception {
+        reconfigServer("server_testcookiename.xml");
+        Expectations expectations = new Expectations();
+        expectations.addExpectations(getSuccessfulLoginPageExpectations(ACTION_INVOKE_PROTECTED_RESOURCE));
+
+        String protectedUrl = "http://" + server.getHostname() + ":" + server.getHttpDefaultPort() + "/formlogin/SimpleServlet";
+        Page response = invokeUrl(new WebClient(), protectedUrl); // get back the login page
+        validationUtils.validateResult(response, ACTION_INVOKE_PROTECTED_RESOURCE, expectations);
+        expectations.addExpectations(getSuccessfulProtectedResourceExpectationsForJwtCookie(ACTION_SUBMIT_LOGIN_CREDENTIALS, protectedUrl));
+        expectations.addExpectations(getJwtPrincipalExpectations(ACTION_SUBMIT_LOGIN_CREDENTIALS));
+
+        response = performLogin(response);
+
+        validationUtils.validateResult(response, ACTION_SUBMIT_LOGIN_CREDENTIALS, expectations);
+
+        // htmlunit getCookies seems not to work, but the test app logs them, check them that way
+        String responseStr = response.getWebResponse().getContentAsString();
+        assertTrue("expected cookie easyrider  not found in cookies", responseStr.contains("cookie: easyrider"));
+        assertTrue("expected cookie LtpaToken2  not found in cookies", responseStr.contains("cookie: LtpaToken2"));
+
+    }
+
+    /**
+     * Test the jwtBuilderRef attribute. Specify a nonexistent builderRef,
+     * login should fail and we should get a meaningful error message in the log.
+     * The ltpa cookie is included, but fallback is false, so fallback should not occur.
+     *
+     * @throws Exception
+     */
+    @Mode(TestMode.LITE)
+    @Test
+    @AllowedFFDC({ "com.ibm.websphere.security.jwt.InvalidBuilderException",
+                   "com.ibm.ws.security.jwt.internal.JwtTokenException",
+                   "com.ibm.websphere.security.WSSecurityException" })
+    public void test_invalidBuilderRef_ltpaFallbackFalse() throws Exception {
+        reconfigServer("server_testbadbuilder.xml");
+        ArrayList<String> ignoredErrors = new ArrayList<String>();
+        ignoredErrors.add("CWWKS6201W");
+        ignoredErrors.add("CWWKS6016E");
```

Comment:
I find it helpful to make message strings a constant and include the message key in the variable name. That way I can see both the message ID being used a general idea of what kind of problem we were looking for. In these cases that would be something like `CWWKS6201W_WARN_JWT_SSO_TOKEN_SERVICE_ERROR` and `CWWKS6016E_JWT_NO_SIGNING_KEY_WITH_ERROR`.